homeassistant/components/opower/coordinator.py

```diff
@@ -230,16 +291,90 @@ async def _async_get_cost_reads(
         - hour resolution for past 2 months (if account's read resolution supports it)
         """
 
+        def _apply_estimiated_costs(
```

Comment:
I haven't looked at the code here but at first glance it looks complicated. Similarly to my previous comment I don't like estimating costs in this integration using the previous month bill. I prefer exposing data as provided by the API without modifying it.

Reply:
Just to make it more clear, the name might be a bit misleading in the context of the estimated forecast data. This does estimate the current billing cycle's costs, but mainly it calculates estimate daily/hourly pricing based on the billing cycle data by spreading it out across the cycle. Otherwise the `cost_reads[i:] = finer_cost_reads[j:]` parts completely overwrite and delete the pricing data.